### PR TITLE
Fix TorchBench transformer and bert_pytorch

### PR DESCRIPTION
**Pull request by tanyokwok**

### CODE REVIEW

File: pytorch_blade/src/torch-mlir/lib/Conversion/MhloPasses.cpp [hunk 1]
```diff
@@ -97,12 +95,12 @@ void mlir::torch::createDiscTorchFunctionToTorchBackendPipeline(
   // Please try to keep this list somewhat up to date when adding
   // "optimize hard enough that it works" transformations.
 
-  // Incorporate user annotations and remove signature Python-isms.
-  pm.addPass(createAdjustCallingConventionsPass());
+  // // Incorporate user annotations and remove signature Python-isms.
+  // pm.addPass(createAdjustCallingConventionsPass());
```
Comment (wyzero) [inline pytorch_blade/src/torch-mlir/lib/Conversion/MhloPasses.cpp:99]: unused code?

File: pytorch_blade/src/torch-mlir/lib/Conversion/MhloPasses.cpp [hunk 2]
```diff
@@ -97,12 +95,12 @@ void mlir::torch::createDiscTorchFunctionToTorchBackendPipeline(
...
 
-  // Eliminate the PrimTupleIndexOp generated from the
-  // adjustCallingConventions
-  pm.addNestedPass<func::FuncOp>(createCanonicalizerPass());
+  // // Eliminate the PrimTupleIndexOp generated from the
+  // // adjustCallingConventions
+  // pm.addNestedPass<func::FuncOp>(createCanonicalizerPass());
```
Comment (wyzero) [inline pytorch_blade/src/torch-mlir/lib/Conversion/MhloPasses.cpp:103]: ditto

File: tao_compiler/mlir/disc/transforms/lhlo_legalize_roots_to_loops.cc [hunk 3]
```diff
@@ -1009,7 +1009,8 @@ LogicalResult emitInThreadReduction(
   for (auto root_op : root_ops) {
     if (isRank2RowReduction(root_op)) {
       if (failed(emitPerElementReductionImpl(
-              b, loc, root_op, acc_iter, input_index, yield_values_for_j))) {
+              b, loc, root_op, acc_iter + row_reduction_idx, input_index,
```
Comment (wyzero) [inline tao_compiler/mlir/disc/transforms/lhlo_legalize_roots_to_loops.cc:1012]: nit: better to add UTs?

File: pytorch_blade/src/torch-mlir/lib/Dialect/TorchConversion/Transforms/VerifyMhloBackendContract.cpp [hunk 4]
```diff
@@ -115,7 +116,9 @@ class VerifyMhloBackendContractPass
         if (isa<ToI64Op, FromI64Op, ToF64Op, FromF64Op>(op)) {
           op->getResult(0).replaceAllUsesWith(op->getOpOperand(0).get());
           op->erase();
-          return;
+        }
+        if (isa<cf::AssertOp>(op)) {
```
[the remark this answers is not in the document]
Comment (tanyokwok) [reply pytorch_blade/src/torch-mlir/lib/Dialect/TorchConversion/Transforms/VerifyMhloBackendContract.cpp:121]: Done

File: pytorch_blade/src/torch-mlir/lib/Conversion/MhloPasses.cpp (hunk 1 above)
Comment (tanyokwok) [reply pytorch_blade/src/torch-mlir/lib/Conversion/MhloPasses.cpp]: Done

File: tao_compiler/mlir/disc/transforms/lhlo_legalize_roots_to_loops.cc (hunk 3 above)
Comment (tanyokwok) [reply tao_compiler/mlir/disc/transforms/lhlo_legalize_roots_to_loops.cc:1011]: Done

File: pytorch_blade/src/compiler/mlir/converters/torch_mlir_op_filter.cpp [hunk 5]
```diff
@@ -95,6 +96,8 @@ const std::unordered_set<std::string> &GetTorchMlirWhiteList() {
       "aten::silu",
       "aten::size",
       "aten::slice",
+      "aten::softmax",
```
Comment (Yancey1989) [inline pytorch_blade/src/compiler/mlir/converters/torch_mlir_op_filter.cpp:99]: add softmax e2e ut in `build_pytorch_blade.sh#L63` ?

Comment (tanyokwok) [reply pytorch_blade/src/compiler/mlir/converters/torch_mlir_op_filter.cpp:100]: done

**Review by wyzero — APPROVED** [review]
LGTM
